src/components/App/ListWrapper.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+
+interface listData {
+    filmTitle: string
+}
+interface listState {
+    filmData: any,
+    visibleOverview: string,
+    visibleDetailview: string,
+    isActive: string,
+    filmTitle: string
+}
+const listCaption : any = ["Title","Released","Runtime","Genre","Director","imdbRating"];
+
+class ListWrapper extends Component<listData, listState> {
+    _isMounted = false;
+    constructor(props: any) {
+        super(props);
+        this.filmTitle = this.props.filmTitle || '';
+        this.state = {filmData : {}, visibleOverview: "show", visibleDetailview: "hide", isActive: "none"};
+    }
+
+    getData = (filmTitle: string) => {
+        const makeRequestUrl = "http://www.omdbapi.com/?apikey=a96e4d00&t=" + filmTitle;
+        const response = fetch(makeRequestUrl).then((response) => { return response.json(); });
+        response.then((response: any) => {
+            if(this._isMounted) {
+                this.setState({
+                    filmData: response,
+                });
+            }
+        });
+    }
+    componentDidMount() {
+        this._isMounted = true;
+        this.getData(this.filmTitle);
+    }
+
+    componentWillUnmount() {
+        this._isMounted = false;
+    }
+    renderListDom = (filmData: any) => {
+        let listEle : any= [];
+        let movieOverview : any = [];
+        let movieDetailedview : any = [];
+        let listEleLen : number = 0;
+        let processedNodes : number = 0;
+        let visibleOverview = this.state.visibleOverview;
+        let visibleDetailview = this.state.visibleDetailview;
+        let ele = filmData;
```

Comment:
Useless variable.

Reply:
Initially i have used different logic to construct the list and detail view, Then I have changed that logic and missed to remove some variables.